x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/container/QueryContainer.java

```diff
@@ -134,45 +133,39 @@ public QueryContainer(Query query,
             return emptyList();
         }
 
-        List<Tuple<Integer, Comparator>> sortingColumns = new ArrayList<>(sort.size());
-
-        boolean aggSort = false;
         for (Sort s : sort) {
-            Tuple<Integer, Comparator> tuple = new Tuple<>(Integer.valueOf(-1), null);
-            
             if (s instanceof AggregateSort) {
-                AggregateSort as = (AggregateSort) s;
-                // find the relevant column of each aggregate function
-                AggregateFunction af = as.agg();
-
-                aggSort = true;
-                int atIndex = -1;
-                String id = Expressions.id(af);
-
-                for (int i = 0; i < fields.size(); i++) {
-                    Tuple<FieldExtraction, String> field = fields.get(i);
-                    if (field.v2().equals(id)) {
-                        atIndex = i;
-                        break;
-                    }
-                }
-                if (atIndex == -1) {
-                    throw new SqlIllegalArgumentException("Cannot find backing column for ordering aggregation [{}]", s);
-                }
-                // assemble a comparator for it
-                Comparator comp = s.direction() == Sort.Direction.ASC ? Comparator.naturalOrder() : Comparator.reverseOrder();
-                comp = s.missing() == Sort.Missing.FIRST ? Comparator.nullsFirst(comp) : Comparator.nullsLast(comp);
-
-                tuple = new Tuple<>(Integer.valueOf(atIndex), comp);
+                customSort = Boolean.TRUE;
```

Comment:
We cannot break after the first AggregateSort. Maybe we could break after the last AggregateSort.
If we have:
```
SELECT f1, f2, f3, MAX(f4) as max, MIN(f5) as min
FROM test
GROUP BY f1, f2, f3
ORDER BY f1, max, f2, min, f3
```
we cannot break after `max`, we could break after `min`.

I'd rather leave the fix as is and introduce this optimisation in a separate PR where it's properly tested that it works.
(Needs some carefully chosen data set to test this ordering case)

Reply:
I don't understand. That's a simple loop that, when finds an AggregateSort, will set `customSort` to `TRUE`. It doesn't really matter what's after in the list of `sort`s because it doesn't change the value of `customSort`.
Also, I meant breaking from inside the loop not from the method...

Reply:
Sorry, misunderstood you there, sure we should break once the 1st is found.